src/registry.rs

```diff
@@ -187,11 +187,20 @@ pub struct RegistryState {
     ready: bool,
 }
 
-#[derive(Debug)]
-struct Global {
+#[derive(Debug, Clone, Eq, Hash, PartialEq, Ord, PartialOrd)]
+pub struct Global {
+    /// The numeric name for this global, usable in [RegistryState::bind_specific].
+    pub name: u32,
+    /// The maximum version supported by the compositor for this global.
+    pub version: u32,
     interface: String,
```

Comment:
Seems a bit odd to have two public fields but then one only accessible via a function

Reply:
I did that to allow switching to `Box<str>` instead of `String` later, if desired.